src/templating/templateEngine.js

```diff
@@ -37,11 +37,63 @@ ko.templateEngine.prototype['createJavaScriptEvaluatorBlock'] = function (script
 ko.templateEngine.prototype['makeTemplateSource'] = function(template, templateDocument) {
     // Named template
     if (typeof template == "string") {
+        var namedTemplateCache = this.namedTemplateCache || {};
+
+        // Cache entries last for 10s after the last access
+        var cacheLifetime = 10000;
+
+        // Registers a timer to clean up stale entries from the template cache.
+        function cleanupCache() {
+            // If we do not yet have a cleanup scheduled, schedule one
+            if (!this.templateCacheCleanupTimeoutID) {
+                this.templateCacheCleanupTimeoutID = setTimeout(function() {
+                    this.templateCacheCleanupTimeoutID = null;
+
+                    var cacheEmpty = true;
+                    var currentTime = new Date().getTime();
+
+                    // Walk through the templates, and delete all that have expired
+                    for (var templateName in namedTemplateCache) {
+                        if (namedTemplateCache.hasOwnProperty(templateName)) {
+                            var template = namedTemplateCache[templateName];
+                            if (currentTime - template.lastAccessed > cacheLifetime) {
+                                delete namedTemplateCache[templateName];
+                            } else {
+                                cacheEmpty = false;
+                            }
+                        }
+                    }
+
+                    // If there are remaining cache entries, register a clean up timer
+                    if (!cacheEmpty) {
```

Comment:
As I understood every time `mateTemplateSource` is called cache cleanUp process is scheduled. Why do you schedule once more cleanUp process for entries which were checked for validity just a moment ago? (in this case you have 100500 cleanUp processes)